### PR TITLE
Raise errors instead of assertion on array type checks

### PR DESCRIPTION
**Pull request by niboshi**
`assert` should be avoided for user input checks.

### CODE REVIEW

File: cupy/util.pyx [hunk 1]
```diff
@@ -198,3 +198,14 @@ The interface can change in the future. ...
...
+
+def check_array(obj, *, arg_name):
+    """Checks if the given object is an array.
+
+    This function raises :class:`TypeError` if ``obj`` is not an instance
+    of :type:`cupy.ndarray`\\ .
+    """
+    if not isinstance(obj, cupy.ndarray):
+        raise TypeError(
+            '\'{}\' must be an array, not {}.'.format(arg_name, type(obj)))
```
Comment (emcastillo) [inline cupy/util.pyx]:
```suggestion
            '\'{}\' must be a cupy.ndarray, not {}.'.format(arg_name, type(obj)))
```

**Comment by niboshi** [comment]
Sorry for delay. PTAL

**Comment by emcastillo** [comment]
Jenkins, test this please.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
I can't understand why but failures are related.

**Comment by niboshi** [comment]
Thanks. PTAL.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**